tasks/main.yml

```diff
@@ -109,25 +109,26 @@
 - name: Humio user-instance configuration file
   tags: humio-conf
   copy:
-    content: "{{ humio_config[item.key] | default('# Empty') }}"
-    dest: "/etc/humio/server_user_{{ item.key }}.conf"
-  with_dict: "{{ ansible_local.cpusockets }}"
+    content: "{{ humio_config[item|string] | default('# Empty') }}"
+    dest: "/etc/humio/server_user_{{ item }}.conf"
+  loop: "{{ range(0, humio_processes|int)|list }}"
   notify: Restart Humio
 
 - name: Create Humio SystemD template configuration
   tags: humio-update
   template:
     src: "humio@.service.j2"
-    dest: /etc/systemd/system/humio@.service
+    dest: /etc/systemd/system/humio@{{ item }}.service
```

Comment:
@watsonian I'm afraid we broke the Systemd template here. Instead of explaining them please take a look at https://www.digitalocean.com/community/tutorials/understanding-systemd-units-and-unit-files#creating-instance-units-from-template-unit-files